### PR TITLE
Statement: Add test for StatementDigestAlreadyAnchored

### LINKED ISSUE #320: [C4GT] Statement: Add tests for `StatementDigestAlreadyAnchored`
## Description
Subtask under : https://github.com/cord-network/community/issues/7

New testcase can be tested under `cargo test -p pallet-statement` after adding it in the code.

## Goals

- [ ] Add tests for `StatementDigestAlreadyAnchored` for pallet/statement

## Expected Outcome
- [ ] Test should assert for `StatementDigestAlreadyAnchored` being returned properly in all the possible calls.
- [ ] If there are more than one function returning this error code, all of them should be validated either as one test case or as multiple test cases.
       
## Acceptance Criteria
NA

## Implementation Details
Look at other test cases, and add a test case for the same.
Ex: check method `trying_to_register_statement_by_a_non_delegate_should_fail` which validates error code `UnauthorizedOperation`

## Mockups / Wireframes
NA

---- 

## Product Name
CORD

## Organization Name
Dhiway

## Domain
Blockchain

## Tech Skills Needed
Rust

## Mentor(s)
@amarts 

## Complexity
[Low]

## Category
[Test]

## Sub Category
[Beginner friendly]

### PR DESCRIPTION
**Pull request by Harsh1s**
Fixes #320 

Description:
This test checks functioning of  the `StatementDigestAlreadyAnchored` error-code by calling the `update` function on a statement with the same digest.

### CODE REVIEW

File: pallets/statement/src/tests.rs [hunk 1]
```diff
@@ -878,3 +878,83 @@ fn trying_to_restore_a_revoked_statement_by_a_non_delegate_should_fail() {
...
+
+		assert_ok!(Space::approve(RawOrigin::Root.into(), space_id.clone(), capacity));
+
+		assert_ok!(Schema::create(
+			DoubleOrigin(author.clone(), creator.clone()).into(),
+			schema.clone(),
+			authorization_id.clone()
+		));
+
+		assert_ok!(Space::add_delegate(
```
Comment (vatsa287) [inline pallets/statement/src/tests.rs]: Since creation of chain space already adds the `creator` as a delegate, why is this required?

Comment (Harsh1s) [reply pallets/statement/src/tests.rs]: Oh I did it by mistake, thanks for pointing it out. I'll fix it up.
